draft-ietf-anima-brski-prm.md

```diff
@@ -1386,17 +1433,17 @@ In addition, the following processing SHALL be performed for PVR contained in RV
 
```

Comment:
above, ll. 1384f:

> Note: Correspond here relates to the case that a single registrar LDevID certificate is used or that different registrar LDevID certificates are used, which are issued by the same CA.

Unclear what is meant by this. My understanding is that the `agent-provided-proximity-registrar-cert` field MUST contain the registrar LDevID certificate, which MUST corresponds==match the certificate and credentials used to sign the RVR.

What is that note trying to say?

Reply:
The thoughts we had were twofold:

1.  if a registrar-agent is in contact with a registrar to get its registrar certificate and provides this information to the pledge in the PVR. The pledge includes the information in the PVR. Next day the registrar-agent connects to a different registrar in the same domain and provides the PVR and RVR. The MASA can now see that there is a mismatch between the initially provided registrar certificate and the one contained in the RVR. Both may be valid as issued by the domain CA, but different instances. This MASA may not be the best one to judge if this is right or wrong, but at least could detect that.
2. If a registrar uses a different certificate for signing the RVR as used in the TLS layer for server authentication, there is a difference between the two certificates. Both are valid and issued by the domain CA. 

Both points were the reason to put correspond and not MUST in the text.

Reply:
So if either of these things happen, the Agent-Proximity Assertion would still hold and the MASA would still assert `agent-proximity`?

Draft 12 also has this statement for countersigning the Voucher:

> The registrar MUST use the same registrar EE credentials used for authentication in the TLS handshake to authenticate towards the Registrar-Agent. This has some operational implications when the registrar may be part of a scalable framework as described in [[I-D.richardson-anima-registrar-considerations](https://www.ietf.org/archive/id/draft-ietf-anima-brski-prm-12.html#I-D.richardson-anima-registrar-considerations)], [Section 1.3.1](https://datatracker.ietf.org/doc/html/draft-richardson-anima-registrar-considerations-08#section-1.3.1).

To me, this would not be necessary if only same-domain is checked; feels a bit contradictory...

Reply:
With my current understanding, I changed it to:

> * the `agent-provided-proximity-registrar-cert` field contains an LDevID certificate that belongs to the same domain as the LDevID certificate/credential used to sign the RVR;
  note that this check allows for installations with multiple domain registrars and for registrar LDevID certificate renewal while PVRs are collected by the Registrar-Agent;
  in many installations with a single registrar the contained certificate is identical to the sigining certificate

The registrar LDevID certificate renewal is the use case I can imagine behind "uses a different certificate for signing the RVR as used in the TLS layer for server authentication", given that otherwise these certs MUST be the same given the countersigning statement (which BTW I moved to Section 6.1 to be applied as global rule).

Reply:
Further added this note after the statements on countersigning the Voucher:

> Note that the credential needs to be the same as used for server authentication in the TLS session with the Registrar-Agent receiving this artifact (see Section 6.1).

Reply:
I like the abstraction. In general that addresses the responsibility of the MASA better, as it does not necessarily need to check, how the operator organized his domain in terms of numbers of registrars.

Reply:
Done with the proposed addition